instrumentation/delayed_job/test/test_helper.rb

```diff
@@ -7,8 +7,13 @@
 require 'bundler/setup'
 Bundler.require(:default, :development, :test)
 
+# These are dependencies that delayed job assumes are already loaded
+# We are compensating for that here in this test... that is a smell
+# NoMethodError: undefined method `extract_options!' for [#<ActiveJobPayload:0x0000000108bf5d48>, {}]:Array
+# delayed_job-4.1.11/lib/delayed/backend/job_preparer.rb:7:in `initialize'0
+require 'active_support/core_ext/array'
```

Comment:
Any reason to not just require `delayed_job` itself here?

Reply:
`delayed_job` is loaded by bundler therefore it is not necessary to load it twice.

Reply:
if required properly, it would load this https://github.com/collectiveidea/delayed_job/blob/b66bb64437a8606a414a390531ac73108911434e/lib/delayed_job.rb#L11 and that would load https://github.com/collectiveidea/delayed_job/blob/b66bb64437a8606a414a390531ac73108911434e/lib/delayed/backend/job_preparer.rb#L1

:thinking:

Reply:
That is very curious isn't it! 

Let me try this again and see what I can figure out

Reply:
Interesting in that it fails for Rails 7.0 but not for 7.1 🤔

Reply:
The curious part for me here is that the backtrace clearly states its raised in JobPreparer, which is what requires the file.

Reply:
Explicitly requiring `delayed_job` and still get this error.

Reply:
https://github.com/open-telemetry/opentelemetry-ruby-contrib/actions/runs/6443662571/job/17495853381?pr=685